src/commands/test.rs

```diff
@@ -129,190 +128,226 @@ impl Command {
+                        let base_token_address =
+                            Some(l2_provider.get_base_token_l1_address().await?);
+                        println!("finalize withdrawal");
+                        wait_for_finalize_withdrawal(withdraw_hash, &l2_provider).await;
+                        arc_zk_wallet.finalize_withdraw(withdraw_hash).await?;
+                        display_balance(base_token_address, &arc_zk_wallet, true, true).await?;
                         println!("{}", "#".repeat(64));
+                        // End Withdrawal
                     }
-                    // End Deposit from rich wallet to rich wallet
 
-                    // Begin Transfer from rich wallet to each wallet
-                    println!("{}", "#".repeat(64));
-                    println!(
-                        "{} Transfer from {} wallet to {} wallet.",
-                        "[L2->L2]".bold().bright_cyan().on_black(),
-                        "rich".bold().red().on_black(),
-                        "each".bold().blue().on_black()
+                    if reruns_wanted != 0 {
+                        reruns += 1;
+                    }
+                    current_reruns += 1;
+
+                    let mut spinner = Spinner::new(
+                        spinners::Dots,
+                        format!("Waiting for {sleep_secs} second(s)"),
+                        Color::Blue,
                     );
+                    sleep(time::Duration::from_secs(sleep_secs));
+                    spinner.success(&format!("Rerun {current_reruns} finished"));
+                }
+                Ok(())
+            }
+            Command::GasScenario {
+                tps,
+                amount,
+                rounds,
+            } => {
+                // TPS, at the moment the calculation is performed with the following conditions
+                // - Don't take deposits into account
+                // - 1 transaction from the rich wallet to each random wallet
+                // - 1 transaction from each random wallet to the rich wallet
+                // - sleep 30 seconds. This step is made to simulate the number of transactions per secondzks
+                //  - (disclaimer) this tps is not the same tps of the chain, here we are sending transactions, the chain is processing transactions
+                //  - (disclaimer) moreover, the "downtime" of the deposit transaction is not taken into account. This reduces the actual tps.
+                //  - the only variable is the amount of random wallets
+                //  - taking into account we will have 2*number_of_wallets transactions
+                //  - the number_of_wallets is calculated as follows: 2*number_of_wallets [txs] = tps [tx/s] * 1 [s]
+                //  - so number_of_wallets = (tps+1)/2. The +1 is to have an even number from an odd number. An even number of tps is preffered.
+                let number_of_wallets = ((tps + 1) / 2).try_into()?;
```

Comment:
Yes, maybe tps is not the best way, maybe a flag called `tpr` = transactions per run . And the idea of `tps+1` is to round the number of wallets up, for example: 3/2 would be 1 but if we add 1 it would be 4/2.

I will be checking it.

Reply:
changed tps with tpr